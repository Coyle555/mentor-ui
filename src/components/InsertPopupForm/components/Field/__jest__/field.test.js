import React from 'react';
import { Field } from '../index';
import renderer from 'react-test-renderer';

test('Default render of the field', () => {
	const tree = renderer.create(<Field />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering field that can go left', () => {
	const handleGoingLeft = jest.fn();

	const tree = renderer.create(
		<Field canGoLeft={true} handleGoingLeft={handleGoingLeft} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering field that can go right', () => {
	const handleGoingRight = jest.fn();

	const tree = renderer.create(
		<Field 
			canGoLeft={false}
			canGoRight={true}
			handleGoingRight={handleGoingRight}
		/>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering field that can go left and right', () => {
	const tree = renderer.create(<Field canGoLeft={true} canGoRight={true} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering a field that can be submitted', () => {
	const _onSubmit = jest.fn();
	const tree = renderer.create(<Field canSubmit={true} _onSubmit={_onSubmit} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering a field with a string value', () => {
	const Input = (props) => <input type="text">{props.value}</input>;

	const tree = renderer.create(
		<Field
			InputComponent={<Input />}
			value="foo"
		/>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering a field with a false valid linked value', () => {
	const Input = (props) => <input {...props} type="text" />;

	const tree = renderer.create(
		<Field
			InputComponent={<Input />}
			link={{ valid: false }}
		/>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering a field with a valid linked value', () => {
	const Input = (props) => <input {...props} type="text" />;

	const tree = renderer.create(
		<Field
			InputComponent={<Input />}
			link={{ valid: true }}
		/>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering field with a onLink callback', () => {
	const onLink = jest.fn(() => ({ value: 'test' }));
	const Input = (props) => <input {...props} type="text" />;

	const tree = renderer.create(
		<Field
			InputComponent={<Input />}
			link={{
				valid: true,
				onLink
			}}
		/>
	).toJSON();

	expect(onLink).toHaveBeenCalled();
	expect(tree).toMatchSnapshot();
});

test('Rendering field with a onLink callback and a link value', () => {
	const onLink = jest.fn((val) => ({}));
	const Input = (props) => <input {...props} type="text" />;

	const tree = renderer.create(
		<Field
			InputComponent={<Input />}
			link={{
				valid: true,
				onLink,
				value: 'foo'
			}}
		/>
	);

	expect(onLink).toHaveBeenCalledWith('foo');
});
