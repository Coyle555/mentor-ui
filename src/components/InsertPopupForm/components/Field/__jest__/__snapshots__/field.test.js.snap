// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render of the field 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  />
  <div
    className="col-2"
  />
</div>
`;

exports[`Rendering a field that can be submitted 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  />
  <div
    className="col-2"
  >
    <button
      className="nav-btn nav-btn-submit"
      onClick={[MockFunction]}
    >
      <i
        className="far fa-check fa-2x"
      />
      <br />
      Submit
    </button>
  </div>
</div>
`;

exports[`Rendering a field with a false valid linked value 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  >
    <input
      data-testid="field-input"
      disabled={false}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-2"
  />
</div>
`;

exports[`Rendering a field with a string value 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  >
    <input
      type="text"
    >
      foo
    </input>
  </div>
  <div
    className="col-2"
  />
</div>
`;

exports[`Rendering a field with a valid linked value 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  >
    <input
      data-testid="field-input"
      disabled={false}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-2"
  />
</div>
`;

exports[`Rendering field that can go left 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  >
    <button
      className="nav-btn"
      onClick={[MockFunction]}
    >
      <i
        className="far fa-chevron-left fa-2x"
      />
      <br />
      Back
    </button>
  </div>
  <div
    className="col-8"
  />
  <div
    className="col-2"
  />
</div>
`;

exports[`Rendering field that can go left and right 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  >
    <button
      className="nav-btn"
      onClick={null}
    >
      <i
        className="far fa-chevron-left fa-2x"
      />
      <br />
      Back
    </button>
  </div>
  <div
    className="col-8"
  />
  <div
    className="col-2"
  >
    <button
      className="nav-btn"
      onClick={null}
    >
      <i
        className="far fa-chevron-right fa-2x"
      />
      <br />
      Next
    </button>
  </div>
</div>
`;

exports[`Rendering field that can go right 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  />
  <div
    className="col-2"
  >
    <button
      className="nav-btn"
      onClick={[MockFunction]}
    >
      <i
        className="far fa-chevron-right fa-2x"
      />
      <br />
      Next
    </button>
  </div>
</div>
`;

exports[`Rendering field with a onLink callback 1`] = `
<div
  className="row"
>
  <div
    className="col-2 text-right"
  />
  <div
    className="col-8"
  >
    <input
      data-testid="field-input"
      disabled={false}
      required={true}
      type="text"
      value="test"
    />
  </div>
  <div
    className="col-2"
  />
</div>
`;
